### ISSUE TITLE
[java] MoreThanOneLogger false positive

### ISSUE THREAD

**Issue by boris-petrov** [pmd/pmd#540]
```java
import java.util.logging.Level;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public final class Application {
	private static final Logger LOGGER = LoggerFactory.getLogger(Application.class);

	private Application() {
	}

	public static void start() {
		java.util.logging.Logger logger = java.util.logging.Logger.getLogger("org.apache.tika.parsers.PDFParser");
		LOGGER.warn("");
		logger.log(Level.WARNING, "");
	}
}
```

This code triggers a `MoreThanOneLogger` warning which is wrong.

PMD version 5.8.1.

**Comment by jsotuyod** [comment]
@boris-petrov thanks for the report.

I'm having trouble understanding why you consider this to be a false positive. You do have 2 logger instances on the same class (a static field and a local variable), and use them both. They are even different logger libs, which points to further inconsistencies.

Why do you consider this usage of different loggers to be legit?

**Comment by boris-petrov** [comment]
@jsotuyod - thanks for the answer!

Well, first of all, the [documentation's example](https://pmd.github.io/pmd-5.8.1/pmd-java/rules/java/logging-java.html#MoreThanOneLogger) contains two loggers for the same `Foo` class - which I can agree is somewhat strange. It doesn't show an example of what I'm doing - i.e. two loggers for different classes (ignore the fact that they're from different libs, that's irrelevant) - which I guessed was not the intention of the warning and that's why I logged it.

It's funny that I extracted a local variable for that logger and only then did PMD warn me about the multiple loggers - before that I had the construction of the second logger inline (passing it to another method) and there was no warning.

In my case I have the logger for the Application class that I want to log stuff with and the other logger I need because I am suppressing a specific message which someone else logs through that logger (using `setFilter`). In my case this *definitely* is a false-positive but I guess there isn't much to be done about that on your side if you prefer to have a warning about multiple loggers.

**Comment by jsotuyod** [comment]
Ok, so this seems to be a one-off where you actually want this, having a class log into another's class logger because of further configuration through a log filter. More like a case you would want to manually whitelist by suppressing this particular instance of the report than something PMD should handle differently.

> It's funny that I extracted a local variable for that logger and only then did PMD warn me about the multiple loggers - before that I had the construction of the second logger inline (passing it to another method) and there was no warning.

Yes, that's a known current limitation. Logging rules are bound to get revamped soon as we tackle things such as #457 

I'm closing this then, feel free to comment if believe we should handle this differently.
